docs/proposals/ThumbnailGallery/README.md

```diff
@@ -0,0 +1,60 @@
+# Component Name
+
+The `ThumbnailGallery` component will accept a list of attachments for images
+and display them as thumbnails. This would also have an option to show the
+images in a LightBox component on click of those images.
+
+## Accessibility
+
+Assistive technology should be able to see the images and know the image name
+based on the `name` property of each attachment.
+
+## Responsiveness & Mobile
+
+This component show maximum number of images possible in a line and show the
+count of images left to show.
+
+## Wireframe
+
+[mockup](mockup.png)
+
+## Interface
+
+```
+<ThumbnailGallery
+    attachments={[{
+            name: "filename.png",
+            url: "https://url-to-that-image"
+            type: "image/png"
+            size: 12345
+`   }]}
+    openGalleryOnClick=true|false
```

Comment:
I don't mind the idea of this. What do you think about a different approach, just to have another idea.

Instead of having the lightbox as part of this component, we would instead have an `onClick` prop that would return the image that was clicked. This way the lightbox can still be used outside of the component, but the consumer of the component can also do other things.

As i typed that I also think that a combination of the 2 would also be nice, something like

```
function handleClick() {
  hasGallery && openGalleryLightbox() // opens the gallery
  onClick && onClick({...image}) // returns the image to the consumber.
}

Reply:
I was under the assumption that showing the thumbnail view and opening images in LightBox is a pattern that we want to follow in more than few places in jobber-online. Without the LightBox, I feel it might be simpler to loop over attachments and use FormatFile in the places that we need it.

Reply:
I am not too familiar working with Atlantis. So does a proposal reflect how the complete component should look like or should it be around the first iteration of the component. For example, we were discussing if we should do the `show more` part of this component later and not for the first iteration. @chriscarline

Reply:
@eddysims I am not too familiar working with Atlantis. So does a proposal reflect how the complete component should look like or should it be around the first iteration of the component. For example, we were discussing if we should do the `show more` part of this component later and not for the first iteration. @chriscarline

Reply:
The proposal is more for you to think through the component before you start working on the component. All of these comments are just considerations for you to think about before you start building the component, so that when the time comes we have thought of (hopefully) all the questions.

Reply:
@Pavithrak To the question of whether the `show more` part should be considered here, my answer is yes! 

The reason we want to do a proposal is to think about the component from a number of perspectives and use cases. And, to consider how this fits into what other components we have, or will have soon. A few examples that can really change or determine a component's behaviour could be: 
- whether this gallery will ever include files, or will just be images (this decision will have pro's and con's for both)
- that it should allow "truncation" of its contents (the "show more") -- i.e. you want a way to control how many items the gallery displays.... and what it does when they click on the "more" image.
- anything else we might possible see this kind of component being used for in the medium term. 

Not that these are hard, just that they take a bit of consideration and communication back and forth to make sure the component does what you need, but in a way that builds on what we already have. 

Ideally, these conversations start with designers when they are first approaching the design of the UI. And, it's usually with designers that these wide ranging considerations make sense (should it include other file types? does that limit what we want to do with image galleries? how will it work on mobile, and on phone web browsers, etc.).

We're still learning how to engage with teams on this stuff, so I will follow up for more feedback in person about this process! :)